management/commands/aggregate_subject_trials.py

```diff
@@ -0,0 +1,526 @@
+        file_size : int
+            The file size in bytes.
+        user : str, User
+            The Alyx database user registering the dataset.
+        revision : str, Revision
+            The revision name to use for the dataset.
+
+        Returns
+        -------
+        Dataset
+            The Dataset record for the aggregate table.
+        pathlib.Path
+            The output file path.
+        """
+        if not file_path.exists():
+            raise FileNotFoundError(file_path)
+        assert all((self.output_path, self.subject)), 'subject and output path must be set'
+        # Get or create the dataset
+        dset, is_new = Dataset.objects.get_or_create(
+            name='_ibl_subjectTrials.table.pqt', collection='Subjects', default_dataset=True,
+            dataset_type=DatasetType.objects.get(name='subjectTrials.table'),
+            data_format=DataFormat.objects.get(name='parquet'), object_id=self.subject.id)
+
+        # Check if unchanged; whether new revision is required
+        if revision or not is_new:
+            unchanged = (
+                    file_hash and file_hash == dset.hash
+                    and aggregate_hash == (dset.json or {}).get('aggregate_hash')
+                    and file_size == dset.file_size)
+            if revision or (dset.is_protected and not unchanged):
+                revision = revision or date.today().isoformat()
+                assert dset.revision is None or dset.revision.name != revision, \
+                    f'Unable to overwrite protected dataset with revision "{revision}".'
+                if isinstance(revision, str):  # user may have already passed in revision obj
+                    revision, _ = Revision.objects.get_or_create(name=revision)
+                logger.info('Creating new dataset with "%s" revision', revision.name)
+                # Create new dataset; leave the old untouched (save method handles change of default dataset field)
+                dset = Dataset.objects.create(
+                    name='_ibl_subjectTrials.table.pqt', collection='Subjects', default_dataset=True,
+                    dataset_type=DatasetType.objects.get(name='subjectTrials.table'),
+                    data_format=DataFormat.objects.get(name='parquet'), content_object=self.subject, revision=revision
+                )
+                is_new = True
+
+        # Update fields with file info and move file to new location
+        dset.version = VERSION
+        dset.file_size = file_size
+        dset.hash = file_hash
+        dset.user = LabMember.objects.get(username=user) if isinstance(user, str) else user
+        dset.json = {**(dset.json or {}), 'aggregate_hash': aggregate_hash}
+        logger.info(('Created' if is_new else 'Updated') + ' aggregate dataset with UUID %s', dset.pk)
+        # Validate dataset
+        dset.full_clean()
+        dset.save()
+
+        # FIXME something's wrong here with the order of collections and revisions :(
```

Comment:
Does this comment need addressing or all good now?

Reply:
Yeah I'll remove this comment.